app/vmagent/remotewrite/relabel.go

```diff
@@ -172,23 +177,13 @@ func loadRelabelConfigs() (*relabelConfigs, error) {
 }
 
 type relabelConfigs struct {
-	global *promrelabel.ParsedConfigs
-	perURL []*promrelabel.ParsedConfigs
+	global  *promrelabel.ParsedConfigs
+	perURL  []*promrelabel.ParsedConfigs
+	flagSet bool // whether (global or per-URL) command-line flags is set
 }
 
 func (rcs *relabelConfigs) isSet() bool {
-	if rcs == nil {
-		return false
-	}
-	if rcs.global.Len() > 0 {
-		return true
-	}
-	for _, pc := range rcs.perURL {
-		if pc.Len() > 0 {
-			return true
-		}
-	}
-	return false
+	return rcs.flagSet
```

Comment:
Nit: I don’t see the necessity of introducing `flagSet` to the `relabelConfigs` struct. The fix seems simpler to me by just adjusting the check to:
```
return *relabelConfigPathGlobal != "" || len(*relabelConfigPaths) > 0
```